presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchConnectorModule.java

```diff
@@ -0,0 +1,152 @@
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.deser.std.FromStringDeserializer;
+import com.google.common.base.Splitter;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import io.airlift.json.JsonCodec;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static io.airlift.configuration.ConfigBinder.configBinder;
+import static io.airlift.json.JsonBinder.jsonBinder;
+import static io.airlift.json.JsonCodecBinder.jsonCodecBinder;
+import static java.nio.file.Files.readAllBytes;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Guice module for the Elasticsearch connector.
+ */
+public class ElasticsearchConnectorModule
+        implements Module
+{
+    @Override
+    public void configure(Binder binder)
+    {
+        binder.bind(ElasticsearchConnector.class).in(Scopes.SINGLETON);
+        binder.bind(ElasticsearchMetadata.class).in(Scopes.SINGLETON);
+        binder.bind(ElasticsearchSplitManager.class).in(Scopes.SINGLETON);
+        binder.bind(ElasticsearchRecordSetProvider.class).in(Scopes.SINGLETON);
+
+        configBinder(binder).bindConfig(ElasticsearchConnectorConfig.class);
+
+        jsonBinder(binder).addDeserializerBinding(Type.class).to(TypeDeserializer.class);
+        jsonCodecBinder(binder).bindJsonCodec(ElasticsearchTableDescription.class);
+
+        binder.install(new DecoderModule());
+    }
+
+    @Singleton
+    @Provides
+    public static ElasticsearchClient createElasticsearchClient(ElasticsearchConnectorConfig config, JsonCodec<ElasticsearchTableDescription> tableDescriptionCodec)
+    {
+        ImmutableMap.Builder<SchemaTableName, ElasticsearchTableDescription> builder = ImmutableMap.builder();
+        try {
+            for (File file : listFiles(config.getTableDescriptionDir())) {
+                if (file.isFile() && file.getName().endsWith(".json")) {
+                    ElasticsearchTableDescription table = tableDescriptionCodec.fromJson(readAllBytes(file.toPath()));
+                    String schemaName = firstNonNull(table.getSchemaName(), config.getDefaultSchema());
+                    builder.put(new SchemaTableName(schemaName, table.getTableName()), table);
+                }
+            }
+
+            Map<SchemaTableName, ElasticsearchTableDescription> tableDefinitions = builder.build();
```

Comment:
You are doing two things in this method, in the first part you create `tableDefinitions`, and then in the second part below you create the `ElasticsearchClient`. Please extract a method for the second part, that will make this method more readable.

Reply:
will do it